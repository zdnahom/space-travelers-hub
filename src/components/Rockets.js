import { useSelector, useDispatch } from 'react-redux';
import { rocketStatus } from '../redux/features/rockets/rocketsSlice';

const Rockets = () => {
  const { rockets, isLoading } = useSelector((store) => store.rockets);
  const dispatch = useDispatch();

  return (
    <div className="rockets-container">
      {isLoading ? (
        <div className="loading">Loading...</div>
      ) : (
        <div>
          {rockets.map((rocket) => (
            <div key={rocket.id} className="rockets">
              <img src={rocket.image} alt={rocket.name} />
              <div className="rockets-description">
                <h2>{rocket.name}</h2>
                <div className="reserved-para">
                  <span className={rocket.reserved ? 'reserved' : ''}>
                    {rocket.reserved ? 'Reserved' : ''}
                  </span>
                  <p className="rocket-para">{rocket.description}</p>
                </div>
                <button
                  type="button"
                  className={rocket.reserved ? 'cancel-reservation' : 'reserve-rocket'}
                  onClick={() => dispatch(rocketStatus(rocket.id))}
                >
                  {rocket.reserved ? 'Cancel Reservation' : 'Reserve Rocket'}
                </button>
              </div>
            </div>
          ))}
        </div>
      )}
    </div>
  );
};

export default Rockets;
