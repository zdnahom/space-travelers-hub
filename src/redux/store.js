import { configureStore } from '@reduxjs/toolkit';
import missionsSlice from './features/missions/missionsSlice';

const store = configureStore({
  reducer: {
    missions: missionsSlice,
  },
});
export default store;
