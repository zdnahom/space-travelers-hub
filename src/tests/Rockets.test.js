import { render, cleanup } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/store';
import '@testing-library/jest-dom';
import Rockets from '../components/Rockets';

afterEach(cleanup);

describe('Rockets component test', () => {
  it('It renders', () => {
    const { container } = render(
      <Provider store={store}>
        <Rockets />
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });
});
