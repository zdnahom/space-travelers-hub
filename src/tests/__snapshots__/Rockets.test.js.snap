// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rockets component test It renders 1`] = `
<div>
  <div
    class="rockets-container"
  >
    <div />
  </div>
</div>
`;
